### PR TITLE
Fix improper k8s to pattern conversion

### PR DESCRIPTION
**Pull request by Revolyssup**
Signed-off-by: Ashish Tiwari <ashishjaitiwari15112000@gmail.com>

**Description**

This PR fixes #
Earlier we were ignoring any fields other than `spec` from k8s manifest(except for few GroupVersion).
**Notes for Reviewers**

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [ ] Yes, I signed my commits.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -65,10 +65,10 @@ require (
 	google.golang.org/protobuf v1.28.0
 	gopkg.in/yaml.v2 v2.4.0
 	gorm.io/gorm v1.23.5
-	k8s.io/api v0.24.2
-	k8s.io/apiextensions-apiserver v0.24.0
-	k8s.io/apimachinery v0.24.2
-	k8s.io/client-go v0.24.2
+	k8s.io/api v0.23.5
```
[the remark this answers is not in the document]
Comment (leecalcote) [reply go.mod]: gotcha.

**Review by leecalcote — APPROVED** [review]
@sudo-NithishKarthik, @gr455, we'll await your review before proceeding.

**Comment by leecalcote** [comment]
@alphaX86 help a fellow contributor out (if you would be so kind) and reconcile these failed unit tests. Note: the CPX adapter has been deprecated.

**Review by alphaX86 — APPROVED** [review]
LGTM. The failing unit tests will be handled and I'll raise a PR by this week to get sorted

**Pull request merged**